controllers/appmesh/virtualservice_controller.go

```diff
@@ -75,6 +78,7 @@ func (r *virtualServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}); err != nil {
 		return err
 	}
+	r.recorder = mgr.GetEventRecorderFor("VirtualService")
```

Comment:
This r.recorder field is part of the virtualServiceReconciler and we initiate all the fields when we get reference to this virtualServiceReconciler, like here
```
vsReconciler := appmeshcontroller.NewVirtualServiceReconciler(mgr.GetClient(), finalizerManager, referencesIndexer, vsResManager, ctrl.Log.WithName("controllers").WithName("VirtualService"))

func NewVirtualServiceReconciler(k8sClient client.Client, finalizerManager k8s.FinalizerManager, referencesIndexer references.ObjectReferenceIndexer, vsResManager virtualservice.ResourceManager, log logr.Logger, recorder record.EventRecorder) *virtualServiceReconciler {
	return &virtualServiceReconciler{
		k8sClient:                             k8sClient,
		finalizerManager:                      finalizerManager,
		referencesIndexer:                     referencesIndexer,
		vsResManager:                          vsResManager,
		enqueueRequestsForMeshEvents:          virtualservice.NewEnqueueRequestsForMeshEvents(k8sClient, log),
		enqueueRequestsForVirtualNodeEvents:   virtualservice.NewEnqueueRequestsForVirtualNodeEvents(referencesIndexer, log),
		enqueueRequestsForVirtualRouterEvents: virtualservice.NewEnqueueRequestsForVirtualRouterEvents(referencesIndexer, log),
		log:                                   log,
	}
}

Similarly we could also pass in recorder to this function
vsReconciler := appmeshcontroller.NewVirtualServiceReconciler(mgr.GetClient(), finalizerManager, referencesIndexer, vsResManager, ctrl.Log.WithName("controllers").WithName("VirtualService"), mgr.GetEventRecorderFor("VirtualService"))
```
Would like to know your thoughts, just to have a consistent pattern, suggesting this approach. Will be easier to track.

Reply:
Agreed, much easier to track. Will revise soon!